torch/csrc/jit/tensorexpr/ir_mutator.cpp

```diff
@@ -271,16 +271,25 @@ const Expr* IRMutator::mutate(const RoundOff* v) {
 }
 
 const Expr* IRMutator::mutate(const ReduceOp* v) {
-  auto accum = v->accumulator().node()->accept_mutator(this);
-  Stmt* init = v->initializer()->accept_mutator(this);
+  const Expr* init = v->initializer()->accept_mutator(this);
   auto body = v->body().node()->accept_mutator(this);
 
+  std::vector<const Expr*> new_output_args;
+  std::vector<const Var*> new_reduce_args;
+  for (auto* e : v->output_args()) {
+    new_output_args.push_back(e->accept_mutator(this));
+  }
+  for (auto* r : v->reduce_args()) {
+    new_reduce_args.push_back(static_cast<const Var*>(r->accept_mutator(this)));
+  }
+
   return new ReduceOp(
-      ExprHandle(accum),
+      v->accumulator(),
```

Comment:
Can Buf* be mutated? I assumed not (are they Exprs?)

Reply:
I think in theory it can be. Note that even though `Buf*` itself is immutable as an `Expr*`, nothing prevents a mutator to create another `Expr*` and return it instead of the original `Buf*` - that's basically how all mutators work on expressions now. For instance, we have a default mutator for `Var*`. I wouldn't expect a default mutator of accumulator to be that useful though - so I agree with you here that it questionable whether we want to "mutate" it or not. In the case of visitor though I do think that we need to go inside.